dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -72,40 +74,42 @@
 class ClassDumper {
   private static final Logger logger = Logger.getLogger(ClassDumper.class.getName());
 
-  private final ImmutableList<Path> inputClassPath;
+  private final ImmutableList<ClassPathEntry> inputClassPath;
   private final FixedSizeClassPathRepository classRepository;
   private final ClassLoader extensionClassLoader;
-  private final ImmutableSetMultimap<Path, String> jarFileToClassFileNames;
-  private final ImmutableListMultimap<String, Path> classFileNameToJarFiles;
+  private final ImmutableSetMultimap<ClassPathEntry, String> classPathEntryToClassFileNames;
+  private final ImmutableListMultimap<String, ClassPathEntry> classFileNameToClassPathEntry;
 
-  private static FixedSizeClassPathRepository createClassRepository(List<Path> paths) {
-    ClassPath classPath = new LinkageCheckClassPath(paths);
+  private static FixedSizeClassPathRepository createClassRepository(List<ClassPathEntry> entries) {
+    ClassPath classPath = new LinkageCheckClassPath(entries);
     return new FixedSizeClassPathRepository(classPath);
   }
 
-  static ClassDumper create(List<Path> jarPaths) throws IOException {
+  static ClassDumper create(List<ClassPathEntry> entries) throws IOException {
     ClassLoader systemClassLoader = ClassLoader.getSystemClassLoader();
     ClassLoader extensionClassLoader = systemClassLoader.getParent();
 
     ImmutableList<Path> unreadableFiles =
-        jarPaths.stream()
+        entries.stream()
+            .map(ClassPathEntry::getPath)
```

Comment:
why doesn't getPath return a Path? Should it?

Reply:
It's because the user of `getPath()` (`LinkageCheckClassPath` and its superclass `org.apache.bcel.util.ClassPath`) expects String.

https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/1312/files#r398210643

Reply:
They could convert to a string if that's what they want. It's surprising to have the method name inconsistent with the return type.